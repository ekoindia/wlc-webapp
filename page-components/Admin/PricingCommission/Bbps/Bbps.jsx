import { Flex, Radio, RadioGroup, Text } from "@chakra-ui/react";
import { Button, Input } from "components";
import { useState } from "react";

/**
 * A <Bbps> component
 * TODO: Write more description here
 * @param 	{object}	prop	Properties passed to the component
 * @param	{string}	prop.prop1	TODO: Property description.
 * @param	{...*}	rest	Rest of the props passed to this component.
 * @example	`<Bbps></Bbps>` TODO: Fix example
 */
const Bbps = () => {
	const [file, setFile] = useState(null);
	console.log("file", file);
	const modeObj = {
		0: "Online",
		1: "Offline",
	};
	const handleFileUpload = async (event) => {
		event.preventDefault();

		try {
			if (!file) return;
			const formData = new FormData();
			formData.append("file", file);

			// hitQuery()
			// // const response = await fetch("/api/upload", {
			// // 	method: "POST",
			// // 	body: formData,
			// // });

			// if (response.ok) {
			// 	console.log("File uploaded successfully");
			// } else {
			// 	console.error("Error uploading file");
			// }
		} catch (error) {
			console.error(error);
		}
	};

	return (
		<Flex direction="column" gap="10" fontSize="md">
			<Flex direction="column" gap="2">
				<Text fontWeight="semibold">Select Mode</Text>
				<RadioGroup
					defaultValue="0"
					// value={commissionType}
					// onChange={(value) => setCommissionType(value)}
				>
					<Flex
						direction={{ base: "column", sm: "row" }}
						gap={{ base: "4", md: "16" }}
					>
						{Object.entries(modeObj).map(([key, value]) => (
							<Radio size="lg" key={key} value={key}>
								<Text fontSize={{ base: "sm", sm: "md" }}>
									{value}
								</Text>
							</Radio>
						))}
					</Flex>
				</RadioGroup>
			</Flex>
			<Flex direction="column" gap="2">
				<Text fontWeight="semibold">Download your current pricing</Text>
				{/* <Button variant="link">Download</Button> */}
			</Flex>
			<Flex direction="column" gap="2">
				<Text fontWeight="semibold">Upload your pricing</Text>
				<DropZone {...{ setFile }} />
			</Flex>
			<Button onClick={handleFileUpload} size="md">
				Upload
			</Button>
		</Flex>
	);
};

export default Bbps;

const DropZone = ({ setFile }) => {
	const [inDropZone, setInDropZone] = useState(false);
	console.log("inDropZone", inDropZone);

	const handleFileInputChange = (event) => {
		const file = event.target.files[0];
		const fileUrl = URL.createObjectURL(file);
		setFile(fileUrl);
	};

	const handleDragOver = (event) => {
		event.preventDefault();
		setInDropZone(true);
	};

	const handleDragLeave = (event) => {
		event.preventDefault();
		setInDropZone(false);
	};

	const handleDrop = (event) => {
		event.preventDefault();
		setInDropZone(false);
		const file = event.dataTransfer.files[0];
		const fileUrl = URL.createObjectURL(file);
		setFile(fileUrl);
	};
	return (
		<Flex
			w={{ base: "100%", md: "500px" }}
			h="200px"
			// bg="divider"
			align="center"
			onDragOver={handleDragOver}
			onDragLeave={handleDragLeave}
			onDrop={handleDrop}
			border="2px dashed #d7dbe3"
		>
			<Flex direction="column" align="center" w="100%" gap="4">
				<Input
					hidden
					type="file"
					onChange={handleFileInputChange}
					id="fileInput"
					padding="120px"
					accept=".xls,.xlms" //TODO check on MIME type
				/>
				<Button
					variant="primary"
					size="md"
					onClick={() => document.getElementById("fileInput").click()}
				>
					You can select your file
				</Button>
				<Text fontSize="sm">or drag &amp; drop your file here</Text>
			</Flex>
		</Flex>
	);
};
//.doc,.docx,.xml,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.pdf,
