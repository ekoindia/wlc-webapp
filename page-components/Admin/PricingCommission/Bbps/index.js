export { default as Bbps } from "./Bbps";
