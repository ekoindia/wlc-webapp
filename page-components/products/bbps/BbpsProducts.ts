import { ParamType } from "constants/trxnFramework";
import { BbpsProduct } from "./types";

export const BbpsProducts: BbpsProduct[] = [
	{
		id: "electricity",
		label: "Electricity Bill",
		desc: "Pay monthly electricity dues",
		icon: "electricity",
		url: "/products/bbps/electricity/search",
		searchFields: [
			{
				name: "utility_acc_no",
				label: "Consumer Number",
				parameter_type_id: ParamType.NUMERIC,
				required: true,
				validations: {
					minLength: 10,
					maxLength: 10,
					pattern: /^[0-9]{10}$/,
					message: "Please enter a valid consumer number",
				},
			},
			{
				name: "confirmation_mobile_no",
				label: "Mobile Number",
				parameter_type_id: ParamType.NUMERIC,
				required: true,
				validations: {
					minLength: 10,
					maxLength: 10,
					pattern: /^[6-9]\d{9}$/,
					message: "Please enter a valid mobile number",
				},
			},
		],
		useMockData: true,
	},
	{
		id: "echallan",
		label: "Traffic E-challan",
		desc: "Pay pending traffic challans",
		icon: "directions-car",
		url: "/products/bbps/echallan/search",
		searchFields: [
			{
				name: "utility_acc_no",
				label: "Vehicle Reg No.",
				parameter_type_id: ParamType.TEXT,
				required: true,
				validations: {
					minLength: 10,
					maxLength: 10,
					pattern: /^[A-Z]{2}[0-9]{2}[A-Z]{1,2}[0-9]{4}$/,
					message: "Please enter a valid vehicle registration number",
				},
			},
		],
		useMockData: true,
	},
	{
		id: "corporate-cc",
		label: "Corporate Credit Card",
		desc: "Clear outstanding corporate card dues",
		icon: "creditcard",
		url: "/products/bbps/corporate-cc/search",
		searchFields: [
			{
				name: "credit_card_no",
				label: "Card Number",
				parameter_type_id: ParamType.TEXT,
				required: true,
				validations: {
					pattern:
						/^(?:4\d{12}(?:\d{3})?|(?:5[1-5]\d{14})|(?:222[1-9]\d{12}|22[3-9]\d{13}|2[3-6]\d{14}|27[01]\d{13}|2720\d{12})|3[47]\d{13}|6(?:011|5\d{2})\d{12}|35(?:2[89]|[3-8]\d)\d{12}|6\d{15})$/,
					message: "Please enter a valid credit card number",
				},
			},
			{
				name: "confirmation_mobile_no",
				label: "Mobile Number",
				parameter_type_id: ParamType.NUMERIC,
				required: true,
				validations: {
					minLength: 10,
					maxLength: 10,
					pattern: /^[6-9]\d{9}$/,
					message: "Please enter a valid mobile number",
				},
			},
		],
		useMockData: true,
	},
];
