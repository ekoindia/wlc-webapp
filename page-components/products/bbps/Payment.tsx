import {
	Box,
	Divider,
	Flex,
	Heading,
	Radio,
	RadioGroup,
	Text,
	useToast,
} from "@chakra-ui/react";
import { ActionButtonGroup, Currency, PageTitle } from "components";
import { useContext, useEffect, useState } from "react";
import { InputPintwin } from "tf-components";
import { BbpsContext } from "./context/BbpsContext";
import { PaymentStatusData, PaymentStatusType } from "./context/types";
import { useBbpsApi } from "./hooks/useBbpsApi";
import { useBbpsNavigation } from "./hooks/useBbpsNavigation";
import { paymentStatusMocks } from "./utils/mockData";

export const Payment = () => {
	const nav = useBbpsNavigation();
	const toast = useToast();
	const { state, dispatch } = useContext(BbpsContext);
	const {
		selectedBills,
		totalAmount,
		useMockData,
		mockResponseType,
		selectedProduct,
		searchFormData,
	} = state;
	const [isProcessing, setIsProcessing] = useState(false);
	// Store encoded PinTwin value (e.g. "4040|87") returned from InputPintwin
	const [pintwinEncoded, setPintwinEncoded] = useState<string>("");
	// Track if PIN has 4 digits entered
	const [pinLength, setPinLength] = useState(0);

	// Get API functions
	const { makePayment } = useBbpsApi(selectedProduct);

	// 🛡️ Navigation Guard - Redirect if invalid state
	useEffect(() => {
		if (selectedBills.length === 0) {
			console.warn(
				"[BBPS Payment] No selected bills, redirecting to preview"
			);
			nav.goPreview();
		}
	}, [selectedBills.length]);

	// Early return if invalid state to prevent InputPintwin from loading
	if (selectedBills.length === 0) {
		return (
			<Box
				p={6}
				bg="white"
				borderRadius="10px"
				boxShadow="basic"
				textAlign="center"
			>
				<Heading size="md" mb={4}>
					No bills selected
				</Heading>
				<Text color="gray.600" mb={4}>
					Redirecting to bill selection...
				</Text>
			</Box>
		);
	}

	// Handle mock response type change
	const handleMockResponseTypeChange = (value: PaymentStatusType) => {
		dispatch({
			type: "SET_MOCK_RESPONSE_TYPE",
			responseType: value,
		});
	};

	// Helper function to handle payment status updates
	const handlePaymentStatus = (
		status: PaymentStatusType,
		message: string,
		transactionId?: string,
		timestamp?: string
	): void => {
		const paymentStatus: PaymentStatusData = {
			status,
			message,
			transactionId:
				transactionId ||
				`TXN${Date.now()}${Math.floor(Math.random() * 1000)}`,
			amount: totalAmount,
			timestamp: timestamp || new Date().toISOString(),
			billIds: selectedBills.map((bill) => bill.billid),
			...(status === "failure" ? paymentStatusMocks.failure : {}),
		};

		dispatch({
			type: "SET_PAYMENT_STATUS",
			payload: paymentStatus,
		});

		// Navigate to status page
		dispatch({ type: "SET_CURRENT_STEP", step: "status" });
		nav.goStatus();
	};

	// Handle payment submission
	const handlePayment = async () => {
		setIsProcessing(true);

		// Construct payment request payload
		const paymentRequest = {
			...searchFormData,
			payment_amount: totalAmount,
			payment_amount_breakup: selectedBills.map((bill) => ({
				billid: bill.billid,
				bill_payment_amount: bill.amount,
			})),
			pintwin: pintwinEncoded, // Encoded PIN with key id (e.g. "4040|87")
			// Add other required fields here based on API specification
		};

		// Log the payload only when using mock data (for debugging)
		if (useMockData) {
			console.log("[BBPS MOCK] Final payment payload:", paymentRequest);
			console.log("[BBPS MOCK] Response type:", mockResponseType);
		}

		// Validate that the sum of bill amounts equals total amount
		const sumOfBillAmounts = paymentRequest.payment_amount_breakup.reduce(
			(sum, item) => sum + item.bill_payment_amount,
			0
		);

		if (Math.abs(sumOfBillAmounts - totalAmount) > 0.01) {
			console.error(
				"[BBPS] Payment validation failed: Sum of bill amounts does not match total amount",
				{ sumOfBillAmounts, totalAmount }
			);
			toast({
				title: "Payment validation failed",
				description: "Sum of bill amounts does not match total amount",
				status: "error",
				duration: 5000,
				isClosable: true,
			});
			setIsProcessing(false);
			return;
		}

		try {
			// Make the actual payment API call - pass mockResponseType when in mock mode
			const { data: response, error } = await makePayment(
				paymentRequest,
				useMockData ? mockResponseType : undefined
			);

			console.log("[BBPS] API response:", response, "error:", error);

			if (error || !response) {
				// Handle payment failure
				handlePaymentStatus(
					"failure",
					error || "Payment failed. Please try again."
				);
			} else if (response.status === 0) {
				// Handle payment success
				handlePaymentStatus(
					"success",
					response.message || "Payment processed successfully",
					response.data?.transactionId,
					response.data?.timestamp
				);
			} else if (response.status === 1) {
				// Handle payment failure (status 1)
				handlePaymentStatus(
					"failure",
					response.message || "Payment failed. Please try again.",
					response.data?.transactionId,
					response.data?.timestamp
				);
			} else {
				// Handle payment pending or other status
				handlePaymentStatus(
					"pending",
					response.message || "Payment is being processed",
					response.data?.transactionId,
					response.data?.timestamp
				);
			}
		} catch (error) {
			console.error("[BBPS] Payment error:", error);
			handlePaymentStatus(
				"failure",
				"An unexpected error occurred during payment"
			);
		} finally {
			setIsProcessing(false);
		}
	};

	const buttonConfigList = [
		{
			type: "submit",
			size: "lg",
			label: "Pay",
			loading: isProcessing,
			disabled: pinLength < 4 || isProcessing,
			onClick: handlePayment,
			styles: { h: "64px", w: { base: "100%", md: "200px" } },
		},
		{
			variant: "link",
			label: "Back to Preview",
			onClick: nav.goPreview,
			disabled: isProcessing,
			styles: {
				color: "primary.DEFAULT",
				bg: { base: "white", md: "none" },
				h: { base: "64px", md: "64px" },
				w: { base: "100%", md: "auto" },
				_hover: { textDecoration: "none" },
			},
		},
	];

	return (
		<>
			<PageTitle
				title="Payment Summary"
				subtitle="Confirm your payment"
				onBack={nav.goPreviousStep}
				toolComponent={
					useMockData && (
						<Box
							px={2}
							py={1}
							bg="yellow.100"
							color="yellow.800"
							borderRadius="md"
							fontSize="sm"
							fontWeight="medium"
						>
							Mock Mode
						</Box>
					)
				}
			/>

			<Flex direction="column" gap="4" mx={{ base: "4", md: "0" }}>
				{/* Payment Summary */}
				<Flex
					direction="column"
					gap="4"
					p={6}
					borderRadius="lg"
					bg="white"
					boxShadow="sm"
				>
					<Text fontSize="lg" fontWeight="bold" color="gray.700">
						Selected Bills
					</Text>
					{/* Bills List */}
					{selectedBills.map((bill) => (
						<Flex
							key={bill.billid}
							justify="space-between"
							p={3}
							bg="gray.50"
							borderRadius="md"
						>
							<Text>{bill.label}</Text>
							<Text fontWeight="bold">
								<Currency amount={bill.amount} />
							</Text>
						</Flex>
					))}
					{/* Divider */}
					<Divider />

					{/* Total Amount */}
					<Flex justify="space-between" fontWeight="bold">
						<Text fontSize="lg">Total Amount</Text>
						<Text fontSize="lg" color="primary.600">
							<Currency amount={totalAmount} />
						</Text>
					</Flex>
				</Flex>

				{/* Payment Options (simplified) */}
				<Flex
					direction="column"
					gap="4"
					p={6}
					borderWidth="1px"
					borderRadius="lg"
					bg="white"
				>
					<Heading size="sm" color="gray.700">
						Payment Method
					</Heading>
					<Text fontSize="sm" color="gray.600">
						Payment will be processed from your default wallet
					</Text>
					<InputPintwin
						label="PIN"
						lengthMin={4}
						lengthMax={4}
						required={true}
						pintwinApp={true}
						// useMockData={true}
						onChange={(value, masked) => {
							if (value.includes("|")) {
								setPintwinEncoded(value);
							}
							// Track PIN length based on masked value (which shows actual digit count)
							setPinLength(masked.length);
							if (useMockData) {
								console.log(
									"[BBPS MOCK] Pintwin encoded:",
									value,
									"masked:",
									masked,
									"length:",
									masked.length
								);
							}
						}}
					/>
				</Flex>

				{/* Mock Response Selection (only shown when useMockData is true) */}
				{useMockData && (
					<Flex
						direction="column"
						gap="4"
						p={6}
						borderRadius="lg"
						bg="yellow.50"
					>
						<Heading size="sm" color="yellow.700">
							Mock Response Type
						</Heading>

						<RadioGroup
							onChange={(value) =>
								handleMockResponseTypeChange(
									value as PaymentStatusType
								)
							}
							value={mockResponseType || "success"}
						>
							<Flex gap="4">
								{Object.values(paymentStatusMocks).map(
									(mock) => (
										<Radio
											key={mock.status}
											value={mock.status}
											colorScheme={
												mock.status === "success"
													? "green"
													: mock.status === "failure"
														? "red"
														: "blue"
											}
										>
											<Text textTransform="capitalize">
												{mock.status}
											</Text>
										</Radio>
									)
								)}
							</Flex>
						</RadioGroup>
					</Flex>
				)}

				<ActionButtonGroup
					buttonConfigList={buttonConfigList}
					bg={{ base: "white", md: "transparent" }}
				/>
			</Flex>
		</>
	);
};
