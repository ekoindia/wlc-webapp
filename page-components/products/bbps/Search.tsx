import { Flex } from "@chakra-ui/react";
import { ActionButtonGroup } from "components";
import router from "next/router";
import { useContext, useEffect } from "react";
import { useForm } from "react-hook-form";
import { Form } from "tf-components/Form";
import { BbpsContext } from "./context/BbpsContext";
import { useBbpsApi } from "./hooks/useBbpsApi";
import { useBbpsNavigation } from "./hooks/useBbpsNavigation";
import { BbpsProduct } from "./types";

export const Search = ({ product }: { product: BbpsProduct }) => {
	const { state, dispatch } = useContext(BbpsContext);
	const nav = useBbpsNavigation();
	const { fetchBills, processBillFetchResponse, isLoadingBills } =
		useBbpsApi(product);

	// Set mock data flag based on product configuration
	useEffect(() => {
		if (product?.useMockData !== state.useMockData) {
			dispatch({
				type: "SET_MOCK_DATA_FLAG",
				useMockData: product?.useMockData || false,
			});
		}
	}, [product?.useMockData, state.useMockData, dispatch]);

	const {
		register,
		control,
		handleSubmit,
		formState: { errors, isSubmitting, isValid },
	} = useForm({
		mode: "onChange",
		defaultValues: product.defaultSearchValues || {},
	});

	console.log("[BBPS] Search errors", errors);
	console.log("[BBPS] Search isValid", isValid);

	const onSubmit = async (payload: Record<string, string>) => {
		console.log("[BBPS] onSubmit", payload);

		// Store search form payload in context
		dispatch({ type: "SET_SEARCH_PAYLOAD", payload: payload });

		// Show loading state
		dispatch({ type: "SET_LOADING", value: true });

		try {
			// Fetch bills (will use mock if product.useMockData is true)
			const { data: response, error } = await fetchBills(payload);

			// Handle error case
			if (error || !response) {
				dispatch({
					type: "SET_ERROR",
					message:
						error || "Failed to fetch bills. Please try again.",
				});
				return;
			}

			// Success case: status === 0
			if (response.status === 0 && response.data) {
				// Transform and store the validation response
				const transformedData = processBillFetchResponse(response);

				if (transformedData) {
					dispatch({
						type: "SET_VALIDATION_RESPONSE",
						payload: transformedData,
					});

					console.log("[BBPS] Transformed response", transformedData);
					nav.goPreview();
				} else {
					dispatch({
						type: "SET_ERROR",
						message:
							"No bills found. Please check your input and try again.",
					});
				}
			} else {
				// Error case: status !== 0
				const errorMessage =
					response.message ||
					"Failed to fetch bills. Please check your input and try again.";
				dispatch({ type: "SET_ERROR", message: errorMessage });
			}
		} catch (error) {
			console.error("[BBPS] Error in onSubmit:", error);
			dispatch({
				type: "SET_ERROR",
				message: "An unexpected error occurred. Please try again.",
			});
		} finally {
			// Hide loading state
			dispatch({ type: "SET_LOADING", value: false });
		}
	};

	const buttonConfigList = [
		{
			type: "submit",
			size: "lg",
			label: "Proceed",
			loading: isSubmitting || isLoadingBills,
			disabled: !isValid,
			styles: { h: "64px", w: { base: "100%", md: "200px" } },
		},
		{
			variant: "link",
			label: "Cancel",
			onClick: () => router.back(),
			styles: {
				color: "primary.DEFAULT",
				bg: { base: "white", md: "none" },
				h: { base: "64px", md: "64px" },
				w: { base: "100%", md: "auto" },
				_hover: { textDecoration: "none" },
			},
		},
	];

	return (
		<form onSubmit={handleSubmit(onSubmit)}>
			<Flex
				direction="column"
				px={{ base: "6", md: "8" }}
				pt="6"
				pb="8"
				bg="white"
				gap="8"
				border="card"
				boxShadow="basic"
				borderRadius="10px"
			>
				<Form
					parameter_list={product.searchFields}
					register={register}
					formValues={state.searchFormData}
					control={control}
					errors={errors}
				/>
				<ActionButtonGroup buttonConfigList={buttonConfigList} />
			</Flex>
		</form>
	);
};
