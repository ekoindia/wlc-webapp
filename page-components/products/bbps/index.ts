export { Bbps } from "./Bbps";
export { Payment } from "./Payment";
export { Preview } from "./Preview";
export { Search } from "./Search";
export { Status } from "./Status";
