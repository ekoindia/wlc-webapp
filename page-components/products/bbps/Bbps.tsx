import { InfoTileGrid } from "components";
import { BbpsProducts } from "./BbpsProducts";

export const Bbps = (): JSX.Element => {
	return <InfoTileGrid list={BbpsProducts} />;
};
