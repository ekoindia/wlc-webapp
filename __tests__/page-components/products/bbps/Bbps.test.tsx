import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Bbps } from "page-components/products/bbps/Bbps";
import { BbpsProducts } from "page-components/products/bbps/BbpsProducts";

// Mock the components
jest.mock("components", () => ({
	InfoTileGrid: ({ list }: { list: any[] }) => (
		<div data-testid="info-tile-grid">
			{list.map((item, index) => (
				<div key={index} data-testid={`tile-${item.id}`}>
					{item.label}
				</div>
			))}
		</div>
	),
	PageTitle: ({ title }: { title: string }) => (
		<h1 data-testid="page-title">{title}</h1>
	),
}));

describe("Bbps Component", () => {
	it("renders without error", () => {
		render(<Bbps />);

		expect(screen.getByTestId("page-title")).toBeInTheDocument();
		expect(screen.getByTestId("info-tile-grid")).toBeInTheDocument();
	});

	it("displays correct page title", () => {
		render(<Bbps />);

		expect(screen.getByTestId("page-title")).toHaveTextContent(
			"Bharat Bill Payment System"
		);
	});

	it("renders InfoTileGrid with BbpsProducts", () => {
		render(<Bbps />);

		const tileGrid = screen.getByTestId("info-tile-grid");
		expect(tileGrid).toBeInTheDocument();

		// Check that all products are rendered
		BbpsProducts.forEach((product) => {
			expect(
				screen.getByTestId(`tile-${product.id}`)
			).toBeInTheDocument();
			expect(screen.getByTestId(`tile-${product.id}`)).toHaveTextContent(
				product.label
			);
		});
	});

	it("renders with correct structure", () => {
		const { container } = render(<Bbps />);

		// Check that the component has the expected structure
		expect(container.firstChild).toMatchSnapshot();
	});

	it("displays all BBPS products", () => {
		render(<Bbps />);

		// Verify specific products are displayed
		expect(screen.getByTestId("tile-corporate-cc")).toHaveTextContent(
			"Corporate Credit Card"
		);
		expect(screen.getByTestId("tile-loan")).toHaveTextContent("Loan");
	});

	it("renders consistently on multiple renders", () => {
		const { rerender } = render(<Bbps />);

		expect(screen.getByTestId("page-title")).toHaveTextContent(
			"Bharat Bill Payment System"
		);

		rerender(<Bbps />);

		expect(screen.getByTestId("page-title")).toHaveTextContent(
			"Bharat Bill Payment System"
		);
		expect(screen.getByTestId("info-tile-grid")).toBeInTheDocument();
	});
});
