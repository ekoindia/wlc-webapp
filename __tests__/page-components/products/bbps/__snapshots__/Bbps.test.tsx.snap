// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bbps Component renders with correct structure 1`] = `
<h1
  data-testid="page-title"
>
  Bharat Bill Payment System
</h1>
`;
